sim/src/mechanics/intersection.rs

```diff
@@ -72,6 +72,7 @@ struct State {
 struct SignalState {
     current_stage: usize,
     stage_ends_at: Time,
+    iteration_count: usize,
```

Comment:
The other two variables are reasonably self-explanatory from the name. It's not clear what the iteration is without reading the code later. What if we called this `extensions_count` and had a comment like "The number of times a variable stage has been extended". Or maybe the var name is fine and just the comment would help